include/mitsuba/render/mesh.h

```diff
@@ -14,31 +18,32 @@ template <typename Float, typename Spectrum>
 class MTS_EXPORT_RENDER Mesh : public Shape<Float, Spectrum> {
 public:
     MTS_IMPORT_TYPES()
-    MTS_IMPORT_BASE(Shape, m_mesh)
+    MTS_IMPORT_BASE(Shape, m_mesh, is_emitter, emitter, is_sensor, sensor)
 
+    // Mesh is always stored in single precision
     using InputFloat = float;
     using InputPoint3f  = Point<InputFloat, 3>;
     using InputVector2f = Vector<InputFloat, 2>;
     using InputVector3f = Vector<InputFloat, 3>;
     using InputNormal3f = Normal<InputFloat, 3>;
 
+    using FloatStorage = DynamicBuffer<replace_scalar_t<Float, InputFloat>>;
+
     using typename Base::ScalarSize;
     using typename Base::ScalarIndex;
 
-    using FaceHolder   = std::unique_ptr<uint8_t[]>;
-    using VertexHolder = std::unique_ptr<uint8_t[]>;
-
     /// Create a new mesh with the given vertex and face data structures
-    Mesh(const std::string &name,
-         Struct *vertex_struct, ScalarSize vertex_count,
-         Struct *face_struct,   ScalarSize face_count);
+    Mesh(const std::string &name, ScalarSize vertex_count,
+         ScalarSize face_count, const Properties &props = Properties(),
+         bool has_vertex_normals = false, bool has_vertex_texcoords = false);
```

Comment:
Does it need a `has_vertex_tangents` as well?

Reply:
I'm actually very confused -- didn't we have vertex tangents as a feature in Mitsuba? This is needed to render anisotropic materials properly. Looking through the diffs, I didn't see this being removed either.

Reply:
We never ported this to Mitsuba 2. Should it be part of another PR maybe?

Reply:
Or maybe this should be "implemented" as a vertex attribute?